packages/number-field/src/NumberField.ts

```diff
@@ -560,8 +563,24 @@ export class NumberField extends TextfieldBase {
             : this._numberFormatter;
     }
 
+    protected get valueFormatter(): NumberFormatter {
+        if (this.step && !this._valueFormatter) {
+            this.digitsAfterDecimal =
+                this.step != Math.floor(this.step)
+                    ? this.step.toString().split('.')[1].length
+                    : 0;
+            this._valueFormatter = new NumberFormatter(
+                this.languageResolver.language,
+                { maximumFractionDigits: this.digitsAfterDecimal }
+            );
+        }
+
+        return this._valueFormatter;
+    }
     private _numberFormatter?: NumberFormatter;
     private _numberFormatterFocused?: NumberFormatter;
+    private _valueFormatter!: NumberFormatter;
+    private digitsAfterDecimal: number = 0;
```

Comment:
Do we need to keep this on the class, or can we use cache clearing to trigger the use of `valueFormatter` as a getter and contain this value therein?

Reply:
Yeah, we can totally do that